activestorage/lib/active_storage/attached/one.rb

```diff
@@ -24,16 +24,10 @@ def blank?
     #   person.avatar.attach(io: File.open("/path/to/face.jpg"), filename: "face.jpg", content_type: "image/jpg")
     #   person.avatar.attach(avatar_blob) # ActiveStorage::Blob object
     def attach(attachable)
-      blob_was = blob if attached?
-      blob = create_blob_from(attachable)
+      new_blob = create_blob_from(attachable)
 
-      unless blob == blob_was
-        transaction do
-          detach
-          write_attachment build_attachment(blob: blob)
-        end
-
-        blob_was.purge_later if blob_was && dependent == :purge_later
+      if !attached? || new_blob != blob
+        write_attachment build_attachment(blob: new_blob)
```

Comment:
Should this write a "change" to the accessor? I don't understand how calling `attach` would now ensure the Attachment got uploaded?

Reply:
We can explore implementing `attach` in terms of the writer in the future. I chose to preserve its existing behavior for now. It uses `create_blob_from`, which uploads as necessary.

Reply:
Ah, got it 👍 There's a fair amount of different flows here 😄